tests/test_nonbonded.py

```diff
@@ -411,82 +397,70 @@ def test_exclusion(self):
-                charge_rescale_mask[i][j] = 1 - exc
-                charge_rescale_mask[j][i] = 1 - exc
-
-            lj_rescale_mask = np.ones((N, N))
-            for (i, j), exc in zip(exclusion_idxs, scales[:, 1]):
-                lj_rescale_mask[i][j] = 1 - exc
-                lj_rescale_mask[j][i] = 1 - exc
-
-            ref_u = functools.partial(
-                nonbonded.nonbonded_v3,
-                charge_rescale_mask=charge_rescale_mask,
-                lj_rescale_mask=lj_rescale_mask,
-                beta=beta,
-                cutoff=cutoff,
-                lambda_plane_idxs=lambda_plane_idxs,
-                lambda_offset_idxs=lambda_offset_idxs,
-            )
+        for precision, rtol in [(np.float64, 1e-8), (np.float32, 1e-4)]:
 
             lamb = 0.0
 
-            params = np.stack(
-                [
-                    (np.random.rand(N).astype(np.float64) - 0.5) * np.sqrt(138.935456),  # q
-                    np.random.rand(N).astype(np.float64) / 10.0,  # sig
-                    np.random.rand(N).astype(np.float64),  # eps
-                ],
-                axis=1,
-            )
+            params = prepare_system_params(test_system)
 
             self.compare_forces(test_system, params, box, lamb, ref_u, test_u, rtol, precision=precision)
 
     def test_nonbonded(self):
 
         np.random.seed(4321)
 
+        _, coords, box, _ = builders.build_water_system(3.0)
+        coords = coords / coords.unit
         for size in [33, 231, 1050]:
 
-            _, coords, box, _ = builders.build_water_system(6.2)
-            coords = coords / coords.unit
-            coords = coords[:size]
+            subset = coords[:size]
 
-            N = coords.shape[0]
+            N = subset.shape[0]
 
             lambda_plane_idxs = np.random.randint(low=-2, high=2, size=N, dtype=np.int32)
             lambda_offset_idxs = np.random.randint(low=-2, high=2, size=N, dtype=np.int32)
 
-            for precision, rtol, atol in [(np.float64, 1e-8, 3e-11), (np.float32, 1e-4, 3e-5)]:
-
-                for cutoff in [1.0]:
-                    # E = 0 # DEBUG!
-                    charge_params, ref_potential, test_potential = prepare_water_system(
-                        coords, lambda_plane_idxs, lambda_offset_idxs, p_scale=1.0, cutoff=cutoff
+            for cutoff in [1.0]:
+                # E = 0 # DEBUG!
+                charge_params, ref_potential, test_potential = prepare_water_system(
+                    subset, lambda_plane_idxs, lambda_offset_idxs, p_scale=5.0, cutoff=cutoff
+                )
+                for precision, rtol, atol in [(np.float64, 1e-8, 1e-8), (np.float32, 1e-4, 5e-4)]:
+
+                    self.compare_forces(
+                        subset,
+                        charge_params,
+                        box,
+                        [0.0, 0.1, 0.2],
+                        ref_potential,
+                        test_potential,
```

Comment:
nit: might be nice to use more kwargs here?
```suggestion
                        lambdas=[0.0, 0.1, 0.2],
                        ref_potential=ref_potential,
                        test_potential=test_potential,
```